glue/core/data_derived.py

```diff
@@ -194,3 +194,181 @@ def compute_histogram(self, *args, **kwargs):
         else:
             kwargs['subset_state'] &= self._indices_subset_state
         return self._original_data.compute_histogram(*args, **kwargs)
+
+
+class SlicedData(BaseCartesianData, HubListener):
```

Comment:
I wonder if we might want to have a single class than handles both slicing and indexing 🤔  But ignore for now 😄

Reply:
Okay, sure 👌🏼